### PR TITLE
Eos 18850: ElasticSearch host to incorporate the multi-node cluster 

### PR DESCRIPTION
**Pull request by Pranali04796**
# Backend

## Problem Statement
Story Ref : [EOS-18850](https://jts.seagate.com/browse/EOS-18850)
Linked PR:  [here](https://github.com/Seagate/cortx-utils/pull/260)

## Unit testing on RPM done
<pre>
Yes
</pre>

## Problem Description
<pre>
1. ElasticSearch host to incorporate the multi-node cluster.
</pre>
## Solution
<pre>
1. Use list for storing host to accept multi-hosts that are in a Elasticsearch cluster.
2. Changes added as per version 7.10.0 for timelion API.
</pre>
## Unit Test Cases: 
<pre>
run csm agent after adding changes from this PR and linked one.
</pre>

Test results (single node):
1. Capacity and HW health alerts
capacity and HW status

2.  CSM Audit logs

3. Performance graph

MicrosoftTeams-image

4. Alerts

5. ES and consul status

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: csm/conf/prepare.py [hunk 1]
```diff
@@ -118,7 +118,25 @@ def _get_data_nw_info(self):
...
             Log.error(f"Network Validation failed.{e}")
             raise CsmSetupError(f"Network Validation failed.{e}")
-        return data_nw_private_fqdn
+        return [data_nw_private_fqdn]
+    
+    def _get_es_hosts_info(self):
+    	"""
+        Obtains minion data network info
+
+    	:param machine_id: Host Minion id.
```
Comment (udayan-y21) [inline csm/conf/prepare.py:127]: Minion id is not the part params .. remove it from  method doc.. Please add the description of return value in method doc..  Write proper description.

File: csm/conf/prepare.py [hunk 2]
```diff
@@ -105,7 +105,7 @@ def _set_fqdn_for_nodeid(self):
             node_name = node_data.get(const.NAME)
             Conf.set(const.CSM_GLOBAL_INDEX, f"{const.MAINTENANCE}>{node_name}",f"{hostname}")
 
-    def _get_data_nw_info(self):
+    def _get_consul_info(self):
         """
         Obtains minion data network info.
```
Comment (udayan-y21) [inline csm/conf/prepare.py:111]: Minion id is not the part params .. remove it from  method doc.. Please add the description of return value in method doc..  Write proper description.

File: csm/conf/prepare.py [hunk 3]
```diff
@@ -127,10 +145,11 @@ def _set_db_host_addr(self):
         :param backend: Databased backend. Supports Elasticsearch('es'), Consul ('consul').
```
Comment (udayan-y21) [inline csm/conf/prepare.py:145]: Backend is not the part params .. remove it from  method doc..  Write proper description.

**Review by udayan-y21 — CHANGES_REQUESTED** [review]

File: csm/conf/prepare.py (hunk 1 above)
Comment (Pranali04796) [reply csm/conf/prepare.py]: Updated required changes in by adding proper description.

File: csm/conf/prepare.py (hunk 2 above)
Comment (Pranali04796) [reply csm/conf/prepare.py]: Updated required changes in by adding proper description.

File: csm/conf/prepare.py (hunk 3 above)
Comment (Pranali04796) [reply csm/conf/prepare.py]: Updated required changes in adding proper description for method.

**Review by pawankumarsrivastava — APPROVED** [review]

**Review by udayan-y21 — APPROVED** [review]

**Review by udayan-y21 — APPROVED** [review]

**Comment by kupranay** [comment]
https://seagate-systems.atlassian.net/wiki/spaces/~340452269/pages/361563003/Opendistro+for+Elasticsearch+and+Kibana

Tested

**Pull request merged**